docsk8s/administering/ls-k8s-stack-monitoring.asciidoc

```diff
@@ -1,14 +1,106 @@
 
-An important step to making your environment production ready is to configure stack monitoring. Monitoring logs and metrics data can be sent to an external resource, such as {ess} or {eck}, so that in the event that any components of your environment become unresponsive, your monitoring data is available.
+To enable {logstash-ref}/monitoring-with-metricbeat.html[Stack Monitoring] for {ls}, you need Metricbeat to collect {ls} metrics, {es} to store the metrics and Kibana to view the result.
 
-Follow these steps to configure monitoring:
+Assuming you have installed ECK, the example modifies the link:https://github.com/elastic/cloud-on-k8s/blob/main/config/recipes/beats/stack_monitoring.yaml[recipe] of Beats stack monitoring. The recipe has initiated a production {es} cluster, a monitoring {es} cluster, {filebeat}, {metricbeat}, a production Kibana and a monitoring Kibana. It monitors {es} and Kibana and send metrics to monitoring cluster.
+We use {metricbeat-ref}/configuration-autodiscover.html[autodiscover] to config monitoring for multiple {ls}.
 
-. Step 1
-. Step 2
-. Step 3
\ No newline at end of file
+* <<ls-k8s-monitor-config-metricbeat>>
+* <<ls-k8s-monitor-config-ls>>
+* <<ls-k8s-monitor-kibana>>
+
+[float]
+[[ls-k8s-monitor-config-metricbeat]]
+==== Config Metricbeat
+
+To monitor {ls}, add logstash module to the recipe.
+
+[source,yaml]
+--
+apiVersion: beat.k8s.elastic.co/v1beta1
+kind: Beat
+metadata:
+  name: metricbeat
+spec:
+  type: metricbeat
+  version: 8.4.2
+  elasticsearchRef:
+    name: elasticsearch-monitoring <1>
+  config:
+    metricbeat:
+      autodiscover:
+        providers:
+          - type: kubernetes
+            scope: cluster
+            hints.enabled: true
+            templates:
+              - condition:
+                  contains:
+                    kubernetes.labels.app: ls <2>
+                config:
+                  - module: logstash <3>
+                    metricsets:
+                      - node
+                      - node_stats
+                    period: 10s
+                    hosts: "http://${data.host}:9600"
+                    xpack.enabled: true
+...
+--
+
+<1> {metricbeat} sends metrics to `elasticsearch-monitoring` cluster.
+
+<2> {metricbeat} scans for the pods with label `app: ls` to collect {ls} metrics.
+
+<3> {metricbeat} logstash module calls metric endpoint from port `9600` for every `10` seconds.
+
+[float]
+[[ls-k8s-monitor-config-ls]]
+==== Config {ls}
+
+Add label `app: ls` to `Deployment` for autodiscover.
+
+[source,yaml]
+--
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: logstash
+  labels:
+    app: ls
+...
+--
+
+Provide the `cluster_uuid` of the production {es} cluster to `monitoring.cluster_uuid` in logstash.yml.
```

Comment:
Do we have a link anywhere on how to do that?

Reply:
I added instructions of getting uuid